crates/ide-diagnostics/src/handlers/moved_out_of_ref.rs

```diff
@@ -172,4 +172,32 @@ fn main() {
             "#,
         );
     }
+
+    #[test]
+    fn regression_14966() {
```

Comment:
Not sure where best to put this regression test. The crash happens during mir lowering, so I'm not sure if `hir_ty::tests` is a better place.

cc @HKalbasi

Reply:
So it won't fail even without the fix? Then I think putting it in `mir::eval::tests` will work.

Reply:
It does fail without the patch, it's just that I'm not sure if placing it in diagnostics tests makes sense (it's not directly related to the diagnostics you see). Would `mir::eval::tests` be a better fit?

Reply:
Ah, I thought it is a `hir_ty::tests::regression` test. Assuming that `hir_ty::tests::regression` won't work since it is only crashing in mir lowering, I would prefer `mir::eval::tests` over here, as it would also checks that it doesn't fail to emit mir.

Reply:
Yep, confirmed that, without this fix, crash can be reproduced in `mir::eval::tests` but not in `hir_ty::tests`. I'll place the test in there. Thanks!